src/session/content/message_row/photo.rs

```diff
@@ -0,0 +1,160 @@
+use glib::clone;
+use gtk::{gdk, gio, glib, prelude::*, subclass::prelude::*, CompositeTemplate};
+use tdgrand::{enums::MessageContent, types::File};
+
+use crate::session::chat::{BoxedMessageContent, Message};
+use crate::session::content::{message_row::Media, MessageRow, MessageRowExt};
+use crate::utils::parse_formatted_text;
+use crate::Session;
+
+mod imp {
+    use super::*;
+    use glib::WeakRef;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/com/github/melix99/telegrand/ui/content-message-photo.ui")]
+    pub struct MessagePhoto {
+        pub binding: RefCell<Option<gtk::ExpressionWatch>>,
+        pub handler_id: RefCell<Option<glib::SignalHandlerId>>,
+        pub old_message: WeakRef<glib::Object>,
+        #[template_child]
+        pub media: TemplateChild<Media>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MessagePhoto {
+        const NAME: &'static str = "ContentMessagePhoto";
+        type Type = super::MessagePhoto;
+        type ParentType = MessageRow;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for MessagePhoto {
+        fn constructed(&self, obj: &Self::Type) {
+            self.parent_constructed(obj);
+            obj.connect_message_notify(|obj, _| obj.update_widget());
+        }
+    }
+
+    impl WidgetImpl for MessagePhoto {}
+}
+
+glib::wrapper! {
+    pub struct MessagePhoto(ObjectSubclass<imp::MessagePhoto>)
+        @extends gtk::Widget, MessageRow;
+}
+
+impl MessagePhoto {
+    pub fn new(message: &glib::Object) -> Self {
+        glib::Object::new(&[("message", message)]).expect("Failed to create MessagePhoto")
```

Comment:
```suggestion
        glib::Object::new(&[("message", message)]).expect("Failed to create ContentMessagePhoto")
```

Reply:
Actually, this whole function can be removed completely. It's not used anywhere and already handled by the base type. 
```suggestion
        glib::Object::new(&[("message", message)]).expect("Failed to create MessagePhoto")
```

Reply:
I usually tend to implement a basic API around `GObject::new()`, `GObject::property()` and `GObject::set_property()` and in general I prefer doing that because it also exposes common APIs that are also found in all GObjects in gtk4-rs. Ideally, in the future gtk4-rs could expose a way to generate/expose these APIs automatically, but that's not the case for now.